service/src/main/java/bio/terra/cbas/controllers/ControllerUtils.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.cbas.controllers;
+
+import bio.terra.cbas.dependencies.sam.SamService;
+import bio.terra.cbas.util.BearerTokenFilter;
+import java.util.Optional;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+@Component
+public class ControllerUtils {
+  private final SamService samService;
+
+  @Autowired
+  ControllerUtils(SamService samService) {
+    this.samService = samService;
+  }
+
+  Optional<String> getUserToken() {
+    Object token =
+        RequestContextHolder.currentRequestAttributes()
+            .getAttribute(BearerTokenFilter.ATTRIBUTE_NAME_TOKEN, RequestAttributes.SCOPE_REQUEST);
+    return Optional.ofNullable((String) token);
+  }
+
+  Optional<UserStatusInfo> getSamUser() {
+    return getUserToken().map(samService::getUserStatusInfo);
```

Comment:
FYI the token is almost certainly a JWT, and might have some of these fields available without a secondary call out to Sam, if we can find the right decoder

Reply:
It actually has _more_ fields than the SamUser call... except the user ID, which is (I suspect) the most important 🙃 When we implement 1 hour tokens from Sam we would be talking to Sam anyways though, so I'm not sure if it's terrible to have to talk to them here.